graphrag/logging/factories.py

```diff
@@ -1,18 +1,18 @@
 # Copyright (c) 2024 Microsoft Corporation.
 # Licensed under the MIT License
 
-"""Load a progress reporter."""
+"""Factory functions for creating loggers."""
 
-from .rich import RichProgressReporter
+from .null_progress import NullProgressReporter
+from .print_progress import PrintProgressReporter
+from .rich_progress import RichProgressReporter
 from .types import (
-    NullProgressReporter,
-    PrintProgressReporter,
     ProgressReporter,
     ReporterType,
 )
 
 
-def load_progress_reporter(
+def create_progress_reporter(
```

Comment:
Maybe a silly question but, what's the difference between `logger` and `reporter` ?

Reply:
That's a great question. The way we use it, I don't believe there is a difference. I personally would like to see a move toward a _logger_ naming convention throughout the codebase instead of _reporter_ as it aligns more with the python logging utility. And in line with the zen of python, explicit is better than implicit. I'm open to suggestions though!

Refactoring the code to use logger* names would make this particular PR too large (I tried it) so I instead chose to focus on just code organization for this PR as a first step.

I'd like to transform the factory functions for logging into a proper class instead of just functions to improve/enforce some of the organization and extensibility around using a factory design pattern